logbucket/logbucket.go

```diff
@@ -0,0 +1,244 @@
+package logbucket
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/honeycombio/honeyelb/publisher"
+)
+
+const (
+	// Somewhat arbitrary -- usually we see about 50 objects per hour in
+	// Honeycomb dogfood data.
+	//
+	// Important thing is that the length is capped and does not grow
+	// indefinitely, old objects will not need de-dupe protection as they
+	// will not be returned in the list objects results once an hour has
+	// elapsed.
+	maxProcessedObjects = 2000
+
+	AWSElasticLoadBalancing     = "elasticloadbalancing"
+	AWSApplicationLoadBalancing = "elasticloadbalancingv2"
+	AWSCloudFront               = ""
+	AWSCloudTrail               = "CloudTrail"
+
+	stateFileFormat = "%s-state-%s.json"
+)
+
+type ObjectDownloadParser struct {
+	// The Publisher provides a way for the object downloaded parser to
+	// publish the parsed events to Honeycomb.
+	*publisher.HoneycombPublisher
```

Comment:
Do you want me to split them into separate pieces with channel glue in `main.go`? Or perhaps I might be able to make a followup PR doing this ? I have some changes which rely a bit on this current architecture to support the CloudFront integration and it might be easier to implement that on top of those than to make the changes and then rebase on top of that.

Reply:
followup is fine.